### PR TITLE
planner: simplify plan cache for fast point get 

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #50618

Problem Summary: planner: simplify plan cache for fast point get

### What changed and how does it work?

planner: simplify plan cache for fast point get

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

File: pkg/sessiontxn/txn_context_test.go [hunk 1]
```diff
@@ -505,12 +505,12 @@ func TestTxnContextForPrepareExecute(t *testing.T) {
...
 		tk.MustExec("prepare s from 'select * from t1 where id=1'")
 	})
-	doWithCheckPath(t, se, normalPathRecords, func() {
+	doWithCheckPath(t, se, []string{"assertTxnManagerInCompile", "assertTxnManagerInShortPointGetPlan"}, func() {
 		tk.MustQuery("execute s").Check(testkit.Rows("1 10"))
 	})
 
 	// Test ExecutePreparedStmt
-	doWithCheckPath(t, se, normalPathRecords, func() {
+	doWithCheckPath(t, se, []string{"assertTxnManagerInCompile", "assertTxnManagerInShortPointGetPlan"}, func() {
```
Comment (qw4990) [inline pkg/sessiontxn/txn_context_test.go:513]: Expected, these 2 cases should be able to reuse PointGet executors

Comment (time-and-fate) [reply pkg/sessiontxn/txn_context_test.go:513]: It's their first time execution, they are not reusing executors.
Previously, it won't go into `(a *ExecStmt) PointGet` for the first execution, now it will. Is that expected?

File: pkg/executor/compiler.go [hunk 2]
```diff
@@ -130,16 +126,10 @@ func (c *Compiler) Compile(ctx context.Context, stmtNode ast.StmtNode) (_ *ExecS
...
-	if pointGetPlanShortPathOK {
-		if ep, ok := stmt.Plan.(*plannercore.Execute); ok {
-			if pointPlan, ok := ep.Plan.(*plannercore.PointGetPlan); ok {
-				stmtCtx.SetPlan(stmt.Plan)
-				stmtCtx.SetPlanDigest(preparedObj.NormalizedPlan, preparedObj.PlanDigest)
-				stmt.Plan = pointPlan
-				stmt.PsStmt = preparedObj
-			} else {
-				// invalid the previous cached point plan
-				preparedObj.PointGet.Plan = nil
```
Comment (time-and-fate) [inline pkg/executor/compiler.go:142]: I didn't find any equivalent logic. Is it safe to remove it directly?

File: pkg/planner/core/plan_cache.go [hunk 3]
```diff
@@ -787,27 +787,17 @@ func checkPreparedPriv(sctx sessionctx.Context, stmt *PlanCacheStmt, is infosche
...
+	if staleread.IsStmtStaleness(sctx) {
 		return false
 	}
 	// check auto commit
 	if !IsAutoCommitTxn(sctx.GetSessionVars()) {
 		return false
 	}
 	if stmt.SchemaVersion != is.SchemaMetaVersion() {
-		stmt.PointGet.Plan = nil
-		stmt.PointGet.ColumnInfos = nil
```
Comment (time-and-fate) [inline pkg/planner/core/plan_cache.go:803]: ditto.

File: pkg/executor/compiler.go [hunk 4]
```diff
@@ -130,16 +126,10 @@ func (c *Compiler) Compile(ctx context.Context, stmtNode ast.StmtNode) (_ *ExecS
 		OutputNames:   names,
 	}
 	// Use cached plan if possible.
-	if pointGetPlanShortPathOK {
-		if ep, ok := stmt.Plan.(*plannercore.Execute); ok {
-			if pointPlan, ok := ep.Plan.(*plannercore.PointGetPlan); ok {
-				stmtCtx.SetPlan(stmt.Plan)
-				stmtCtx.SetPlanDigest(preparedObj.NormalizedPlan, preparedObj.PlanDigest)
```
Comment (time-and-fate) [inline pkg/executor/compiler.go:137]: Why is this line removed?

**Review by time-and-fate — APPROVED** [review]
